cpp/include/cuml/cuml_api.h

```diff
@@ -20,13 +20,24 @@
 
 #include <cuda_runtime_api.h>
 
+// Block inclusion of this header when compiling libcuml++.so. If this error is
+// shown during compilation, there is an issue with how the `#include` have
+// been set up. To debug the issue, run `./build.sh cppdocs` and open the page
+// 'cpp/build/html/cuml__api_8h.html' in a browser. This will show which files
+// directly and indirectly include this file. Only files ending in '*_api' or
+// 'cumlHandle' should include this header.
```

Comment:
This comment might be useful to add to the readme/dev guide (if its already there disrefard comment of course)

Reply:
Good suggestion. I added this by the error message because I assumed developers would hit the error, go to the source, and then be able to see the message. I'll add it to the dev guide somewhere.

Reply:
Fixed